csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.shared.cs

```diff
@@ -979,66 +976,50 @@ private void InitWithSessionHandle(IntPtr session, SessionOptions options)
             }
             catch (Exception)
             {
-                if (_nativeHandle != IntPtr.Zero)
-                {
-                    NativeMethods.OrtReleaseSession(_nativeHandle);
-                    _nativeHandle = IntPtr.Zero;
-                }
+                DisposeImpl(true);
                 throw;
             }
 
             _builtInRunOptions = new RunOptions();  // create a default built-in run option, and avoid creating a new one every run() call  
         }
 
 
-        private string GetOutputName(ulong index, out byte[] utf8)
+        private string GetOutputName(ulong index, out IntPtr utf8)
```

Comment:
We are not going back. We are caching utf8 strings in unmanaged memory, so I reworked the function to make a copy directly into unmanaged memory rather than having an intermediate in a form of `byte[]`.